pkg/policy/api/rule.go

```diff
@@ -293,10 +309,28 @@ type PortProtocol struct {
 	Protocol L4Proto `json:"protocol,omitempty"`
 }
 
+// IngressVisibilityRule ensures the ingress traffic to a set of L4 ports is
+// forwarded at L7 for the purpose of logging requests and responses into the
+// access log.
+// Only the traffic to the ports that is allowed by policy at L3 and L4 is
+// visible.
+type IngressVisibilityRule struct {
+	// ToPorts is a list of L4 port/protocol.
+	//
+	// +optional
+	ToPorts []PortProtocol `json:"toPorts,omitempty"`
+
+	// L7Protocol identifies the L7 protocol of the ingress traffic to the
+	// ports. Case insensitive. Accepted values: "http", "kafka".
+	//
+	// +optional
```

Comment:
Is this really optional? Sanitize seems to enforce that this value is either http or kafka.